lightning/src/ln/functional_test_utils.rs

```diff
@@ -60,59 +60,110 @@ pub fn mine_transaction<'a, 'b, 'c, 'd>(node: &'a Node<'b, 'c, 'd>, tx: &Transac
 /// Mine the given transaction at the given height, mining blocks as required to build to that
 /// height
 pub fn confirm_transaction_at<'a, 'b, 'c, 'd>(node: &'a Node<'b, 'c, 'd>, tx: &Transaction, conf_height: u32) {
-	let starting_block = node.best_block_info();
+	let first_connect_height = node.best_block_info().1 + 1;
+	assert!(first_connect_height <= conf_height);
+	if conf_height - first_connect_height >= 1 {
+		connect_blocks(node, conf_height - first_connect_height);
+	}
 	let mut block = Block {
-		header: BlockHeader { version: 0x20000000, prev_blockhash: starting_block.0, merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 },
+		header: BlockHeader { version: 0x20000000, prev_blockhash: node.best_block_hash(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 },
 		txdata: Vec::new(),
 	};
-	let height = starting_block.1 + 1;
-	assert!(height <= conf_height);
-	for _ in height..conf_height {
-		connect_block(node, &block);
-		block = Block {
-			header: BlockHeader { version: 0x20000000, prev_blockhash: block.header.block_hash(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 },
-			txdata: vec![],
-		};
-	}
-
 	for _ in 0..*node.network_chan_count.borrow() { // Make sure we don't end up with channels at the same short id by offsetting by chan_count
 		block.txdata.push(Transaction { version: 0, lock_time: 0, input: Vec::new(), output: Vec::new() });
 	}
 	block.txdata.push(tx.clone());
 	connect_block(node, &block);
 }
 
+/// An enum describing the possible ways we may notify a ChannelManager of a new block
+pub enum ConnectStyle {
+	/// Calls update_best_block first, detecting transactions in the block only after receiving the
+	/// header and height information.
+	BestBlockFirst,
+	/// The same as BestBlockFirst, however when we have multiple blocks to connect, we only
+	/// make a single update_best_block call.
+	BlockSkippingBestBlockFirst,
+	/// Calls transactions_confirmed first, detecting transactions in the block before updating the
+	/// header and height information.
+	TransactionsFirst,
+	/// The same as TransactionsFirst, however when we have multiple blocks to connect, we only
+	/// make a single update_best_block call.
+	BlockSkippingTransactionsFirst,
```

Comment:
I think the `BlockSkipping` variants would be more readable as `BestBlockFirstSkippingBlocks` and `TransactionsFirstSkippingBlocks`. Otherwise, it sounds as if what follows "skipping" is what is being skipped.